core/frontend/src/tile/map/MapLayerSources.ts

```diff
@@ -25,16 +25,37 @@ export enum MapLayerSourceStatus {
   RequireAuth,
 }
 
+/** JSON representation of a map layer source.
+ * @internal
+ */
+interface MapLayerSourceProps extends MapLayerProps
+{
+  /** Indicate if this source definition should be used as a base map. */
+  baseMap?: boolean;
+  /** UserName */
+  userName?: string;
+  /** Password */
+  password?: string;
+}
+
 /** A source for map layers.  These may be catalogued for convenient use by users or applications.
  * @internal
  */
 export class MapLayerSource implements MapLayerProps {
   public subLayers?: MapSubLayerProps[];
 
-  private constructor(public formatId: string, public name: string, public url: string, public baseMap = false, public transparentBackground?: boolean, public maxZoom?: number, public userName?: string, public password?: string) { }
-  public static fromJSON(json: any): MapLayerSource | undefined {
+  private constructor(public formatId: string, public name: string, public url: string, public baseMap = false, public transparentBackground?: boolean, public userName?: string, public password?: string) { }
```

Comment:
nitpick but I hate these constructors that declare half a dozen class members inline - they are convenient once, for the author who gets to type a little less, and inconvenient forever, for everyone else who has to parse them.

Reply:
Given the constructor is private and only called by `fromJson`, consider making it accept MapLayerSourceProps directly so you're not handling default values in two places.